configs/components/puppet.rb

```diff
@@ -235,4 +236,55 @@
   if platform.is_eos?
     pkg.link "#{settings[:sysconfdir]}", "#{settings[:link_sysconfdir]}"
   end
+
+  pkg.install_file "ext/hiera/hiera.yaml", File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
+  pkg.configfile File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
+  pkg.configfile File.join(configdir, 'hiera.yaml')
+
+  unless platform.is_windows?
+    old_hiera = File.join(settings[:puppet_codedir], 'hiera.yaml')
+    cnf_hiera = File.join(configdir, 'hiera.yaml')
+    env_hiera = File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
+    rmv_hiera = File.join(configdir, 'remove_hiera5_files.rm')
+
+    # Pre-Install script to save copies of existing hiera configuration files.
+    # This also creates "marker files to indicate that the files pre-existed."
+    preinstall = <<-PREINST
+# Backup the old hiera configs if present, so that we
+# can drop them back in place if the package manager
+# tries to remove it.
```

Comment:
Can and should we back up and restore `/etc/puppetlabs/code/environments/production/hiera.yaml` ? Ie for users going from 5.0.0 to 5.0.1 ?